File: ReadyTasks/CSharp/Rot13/Rot13/Program.cs
using System;
using System.Linq;

namespace Rot13
{
    class Program
    {
        static string Alphabet = "abcdefghijklmnopqrstuvwxyz";

        static char RotateCharacter(char letter, int rotation)
        {
            int ind = Alphabet.IndexOf(letter);
            ind += rotation;
            ind %= Alphabet.Length;
            return Alphabet[ind];
        }

        static string RotateString(string s, int rotation)
        {
            return String.Concat(s.Select(x => RotateCharacter(x, rotation)));
        }

        static string ROT13(string s)
        {
            return RotateString(s, 13);
        }

        static void Main(string[] args)
        {
            Console.Write("Write string: ");
            string text = Console.ReadLine();

            Console.Write("Rotated string: ");
            Console.WriteLine(ROT13(text));

            Console.Write("Rotated twice string: ");
            Console.WriteLine(ROT13(ROT13(text)));
        }
    }
}

File: ReadyTasks/FSharp/Rot13/Rot13/Program.fs
open System

let rotateLetter (rotation: int) (letter: char) =
    let alphabet = "abcdefghijklmnopqrstuvwxyz"
    let ind = alphabet.IndexOf(letter) + rotation
    let newInd = ind % alphabet.Length
    alphabet.[newInd]

let rotateString (rotation: int) (text: string) =
    text |> String.map (rotateLetter rotation)

let rot13 = rotateString 13

[<EntryPoint>]
let main argv =
    Console.Write("Write text: ")
    let text = Console.ReadLine()

    Console.Write("Rotated text: ")
    Console.WriteLine(text |> rot13)

    Console.Write("Rotated twice text: ")
    Console.WriteLine(text |> rot13 |> rot13)
    0 // return an integer exit code